src/Compilers/Core/Portable/Collections/CachingFactory.cs

```diff
@@ -154,16 +154,16 @@ public void Add(TKey key, TValue value)
             var hash = RuntimeHelpers.GetHashCode(key);
             var idx = hash & mask;
 
-            entries[idx].key = key;
-            entries[idx].value = value;
+            Entries[idx].key = key;
+            Entries[idx].value = value;
         }
 
         public bool TryGetValue(TKey key, [MaybeNullWhen(returnValue: false)] out TValue value)
         {
             int hash = RuntimeHelpers.GetHashCode(key);
             int idx = hash & mask;
 
-            var entries = this.entries;
+            var entries = this.Entries;
```

Comment:
>var entries = this.Entries; [](http://example.com/codeflow?start=12&length=27)

nit: TryGetValue doesn't need to realize the Entries array

Reply:
fair.  but this keeps things exactly the same otherwise.  something we could consider.